aclk/agent_cloud_link.c

```diff
@@ -303,7 +359,27 @@ struct aclk_query *aclk_queue_pop()
 
     this_query = aclk_queue.aclk_query_head;
 
-    if (this_query->run_after > now_realtime_sec()) {
+    // Get rid of the deleted entries
```

Comment:
Deletion in a queue is fast - why are you doing deferred deletion instead?

Reply:
There is code to delete in place. Needs some optimization. Will be done in another issue along with the deferred deletion code (and cleanup of the data structure)